src/web/views/UMM/index.jsx

```diff
@@ -0,0 +1,272 @@
+
+      this.throttled()
+    }
+  }
+
+  handleAddTemplateToDocument(template) {
+    const code = this.state.code + '\n' + template + '\n'
+    this.setState({
+      code: code
+    })
+
+    const editor = document.getElementsByClassName('CodeMirror')[0]
+    
+    if (editor) {
+      const cm = editor.CodeMirror
+      cm.setValue(code)
+
+      setImmediate(() => {
+        this.throttled()
+        
+        this.highlightAndMoveTo(cm, template)
+        const node = ReactDOM.findDOMNode(this.end)
+        node.scrollIntoView({ behavior: "smooth" })
+      })
+    }
+  }
+
+  handleSave() {
+    return axios.post('/umm/blocs', {
+      content: this.state.code
+    })
+    .then(() => {
+      this.setState({
+        hashCode: this.state.code
+      })
+    })
+  }
+
+  handlePreviewLoadingChanged(loading) {
+    this.setState({
+      previewLoading: loading
+    })
+  }
+
+  handleResetRefresh() {
+    this.setState({
+      refresh: false
+    })
+  }
+
+  render() {
+    if (this.state.loading) {
+      return null
+    }
+
+    const classNames = classnames({
+      [style.umm]: true,
+      'bp-umm': true
+    })
+
+    const hasChanged = this.state.hashCode && (this.state.hashCode !== this.state.code)
+
+    return (
+      <ContentWrapper>
+        <PageHeader><span>Universal Message Markdown</span></PageHeader>
+        <table className={classNames}>
+          <tbody>
+            <tr>
+              <td style={{ width: '20%' }}>
+                <Search search={this.state.search} update={::this.handleSearchChanged}/>
+              </td>
+              <td style={{ width: '40%' }}>
+                <Actions 
+                  templates={this.state.templates}
+                  error={this.state.error}
+                  add={::this.handleAddTemplateToDocument} />
+              </td>
+              <td style={{ width: '40%' }}>
+                <Platform 
+                  selected={this.state.selectedPlatform}
+                  platforms={this.state.supportedPlatforms} 
+                  update={::this.handleSelectedPlatformChanged}
+                  save={::this.handleSave}
+                  changed={hasChanged}/>
+              </td>
+            </tr>
+            <tr>
+              <td style={{ width: '20%' }}>
+                <List
+                  blocks={this.state.blocks} 
+                  search={this.state.search}
+                  selected={this.state.selectedBlock}
+                  update={::this.handleSelectedBlockChanged} />
+              </td>
+              <td style={{ width: '40%' }} className={classnames(style.tdCode, 'bp-umm-td-code')}>
+                <Code 
+                  erro={this.state.error}
+                  code={this.state.code}
+                  refresh={this.state.refresh}
+                  resetRefresh={::this.handleResetRefresh}
+                  update={::this.handleDocumentChanged}
+                  setLoading={::this.handlePreviewLoadingChanged} />
```

Comment:
This is reversed react logic. Lower components should never dictate how higher components behave. Instead you should name this: `onLoaded` so that the higher component do whatever he wants with the event